HARK/ConsumptionSaving/ConsRiskyContribModel.py

```diff
@@ -0,0 +1,1768 @@
+        end of the current period and the start of next period. Their order is
+        (0) permanent income shock, (1) transitory income shock, (2) risky
+        asset return.
+    a : float
+        end-of-period risk-free assets.
+    nTil : float
+        end-of-period risky assets.
+    s : float
+        end-of-period income deduction share.
+    dvdm_next : 3D function
+        Next-period's marginal value of riskless assets.
+    dvdn_next : 3D function
+        Next-period's marginal value of risky assets.
+    dvds_next : 3D function
+        Next-period's marginal value of income-deduction share.
+    CRRA : float
+        Coefficient of relative risk aversion.
+    PermGroFac : float
+        Permanent income deterministic growth factor.
+    Rfree : float
+        Risk-free return factor.
+    DiscFac : float
+        Time-preference discount factor.
+    LivPrb : float
+        Survival probability.
+    v_next : 3D function, optional
+        Next-period's value function. The default is None.
+
+    Returns
+    -------
+    np.array
+        Array with end-of-period-value derivatives conditional on next
+        period's shocks. Order
+        (0) riskless assets, (1) risky assets, (2) income deduction share
+        Optionally, the level of end-of-period value is added in position (3).
+
+    """
+
+    temp_fac_A = utilityP(shocks[0] * PermGroFac, CRRA)
+    temp_fac_B = (shocks[0] * PermGroFac) ** (1.0 - CRRA)
+
+    # Find next-period asset balances
+    m_next = m_nrm_next(shocks, a, s, Rfree, PermGroFac)
+    n_next = n_nrm_next(shocks, nTil, s, PermGroFac)
+
+    # Interpolate next-period-value derivatives
+    dvdm_tp1 = dvdm_next(m_next, n_next, s)
+    dvdn_tp1 = dvdn_next(m_next, n_next, s)
+    if shocks[1] == 0:
+        dvds_tp1 = dvds_next(m_next, n_next, s)
+    else:
+        dvds_tp1 = shocks[1] * (dvdn_tp1 - dvdm_tp1) + dvds_next(m_next, n_next, s)
+
+    # Discount next-period-value derivatives to current period
+
+    # Liquid resources
+    end_of_prd_dvda = DiscFac * Rfree * LivPrb * temp_fac_A * dvdm_tp1
+    # Iliquid resources
+    end_of_prd_dvdn = DiscFac * shocks[2] * LivPrb * temp_fac_A * dvdn_tp1
+    # Contribution share
+    end_of_prd_dvds = DiscFac * LivPrb * temp_fac_B * dvds_tp1
+
+    # End of period value function, if needed
+    if v_next is not None:
+        end_of_prd_v = DiscFac * LivPrb * temp_fac_B * v_next(m_next, n_next, s)
+        return np.stack(
+            [end_of_prd_dvda, end_of_prd_dvdn, end_of_prd_dvds, end_of_prd_v]
+        )
+    else:
+        return np.stack([end_of_prd_dvda, end_of_prd_dvdn, end_of_prd_dvds])
+
+
+# %% RiskyContrib solvers
+
+# Consumption stage solver
+def solve_RiskyContrib_Cns(
+    solution_next,
+    ShockDstn,
+    LivPrb,
+    DiscFac,
+    CRRA,
+    Rfree,
+    PermGroFac,
+    BoroCnstArt,
+    aXtraGrid,
+    nNrmGrid,
+    mNrmGrid,
+    ShareGrid,
+    vFuncBool,
+    AdjustPrb,
+    DiscreteShareBool,
+    **unused_params
+):
```

Comment:
No docstring?

Reply:
Added docstrings to all solution functions in https://github.com/econ-ark/HARK/pull/832/commits/824d4f8c9016224e46b7125c0e567e9d694950c4

Thank you!